examples/llm/src/mosaic_gpt.py

```diff
@@ -421,10 +421,12 @@ def forward(self,
                                     seq_len=S,
                                     key_padding_mask=key_padding_mask,
                                     dtype=x.dtype)
+        if self.cfg.attn_impl == 'flash':
+            key_padding_mask = torch.ones_like(input_ids, dtype=torch.bool)
```

Comment:
doesn't this need to actually take into account padding? tokenization on the fly is still supported, as long as you use a valid tokenizer.